src/pybind/mgr/orchestrator_cli/module.py

```diff
@@ -42,6 +42,132 @@ class OrchestratorCli(orchestrator.OrchestratorClientMixin, MgrModule):
+                'summary': '%d devices have fault light turned on' % len(
+                    self.fault),
+                'detail': ['{} fault light enabled'.format(d) for d in self.ident]
+            }
+        self.set_health_checks(h)
+
+    def _get_device_locations(self, dev_id):
+        # type: (str) -> List[orchestrator.DeviceLightLoc]
+        locs = [d['location'] for d in self.get('devices')['devices'] if d['devid'] == dev_id]
+        return [orchestrator.DeviceLightLoc(**l) for l in  sum(locs, [])]
+
+    @_read_cli(prefix='device ls-lights',
+               desc='List currently active device indicator lights')
+    def _device_ls(self):
+        return HandleCommandResult(
+            stdout=json.dumps({
+                'ident': list(self.ident),
+                'fault': list(self.fault)
+                }, indent=4))
+
+    def light_on(self, fault_ident, devid):
+        # type: (str, str) -> HandleCommandResult
+        assert fault_ident in ("fault", "ident")
+        locs = self._get_device_locations(devid)
+        if locs is None:
+            return HandleCommandResult(stderr='device {} not found'.format(devid),
+                                       retval=-errno.ENOENT)
+
+        getattr(self, fault_ident).add(devid)
+        self._save()
+        self._refresh_health()
+        completion = self.blink_device_light(fault_ident, True, locs)
+        self._orchestrator_wait([completion])
+        return HandleCommandResult(stdout=str(completion.result))
+
+    def light_off(self, fault_ident, devid, force):
+        # type: (str, str, bool) -> HandleCommandResult
+        assert fault_ident in ("fault", "ident")
+        locs = self._get_device_locations(devid)
+        if locs is None:
+            return HandleCommandResult(stderr='device {} not found'.format(devid),
+                                       retval=-errno.ENOENT)
+
+        try:
+            completion = self.blink_device_light(fault_ident, False, locs)
+            self._orchestrator_wait([completion])
+
+            if devid in getattr(self, fault_ident):
+                getattr(self, fault_ident).remove(devid)
+                self._save()
+                self._refresh_health()
+            return HandleCommandResult(stdout=str(completion.result))
+
+        except:
+            # There are several reasons the try: block might fail:
+            # 1. the device no longer exist
+            # 2. the device is no longer known to Ceph
+            # 3. the host is not reachable
+            if force and devid in getattr(self, fault_ident):
+                getattr(self, fault_ident).remove(devid)
+                self._save()
+                self._refresh_health()
+            raise
+
+
+    @_write_cli(prefix='device fault-light-on',
+                cmd_args='name=devid,type=CephString',
+                desc='Enable device *fault* light')
+    def _device_fault_on(self, devid):
+        return self.light_on('fault', devid)
```

Comment:
So, the idea was to keep the CLI similar to [`lsmcli`(https://www.systutorials.com/docs/linux/man/1-lsmcli/#lbCA)

Reply:
Why? I haven't used libStorage since a lot of time ago, but it's the kind of command that you always need to look for (`local-disk-fault-led-on`).

I mean, the point in having `ceph device light <kind> <on/off>` is that when you're looking for that if follows top-to-bottom approach (device -> light -> etc). If you're looking for the sub command to make device lights blink, it seems easier to look for `lights` rather than for `fault-lights-on`, and the shell completion will also help (not sure if we are defining completions for Mgr modules).

Reply:
Hmm, I think a single command might be friendlier too.  No need to match lsmcli, which most likely nobody has ever used.

Reply:
Some months ago, I heard some argument that we never should manage the fault light and let the firmware manage the fault led. And instead concentrate on the ident led. Don't remember who came up with this, unfortunately. Maybe @pcuzner ? That would make the cli look like

```bash
ceph device light (on|off) <devid>
ceph device light set-light-type (fault|ident)  # or making this a module setting? 
```

Reply:
Hmm, perhaps.  I think *some* firmwares/drives will do the fault LED for you, but I suspect not all will.  Perhaps it makes sense to make this an optional argument (and default to what a config value says) so that the human doesn't have to remember which one to use for their particular environment/cluster, but in general, long-term, I think what we probably want is for ceph to take over the management of the lights so that the fault LED can be integrated with our own notion of what a fault means?

My suggestion would be to change type to a CephChoices fault|ident with default to ident and make it optional...